zeroconf.py

```diff
@@ -1726,28 +1729,16 @@ def get_network_interfaces():
                     # There are no methods for checking IP address type, so check object type.
                     address = ip.getHostAddress()
                     if isinstance(ip, Inet4Address):
-                        interfaces.append({"inet": address, "inet6": ""})
-                    else:
-                        interfaces.append({"inet": "", "inet6": address})
+                        addresses.append(address)
         except Exception as e:
             return []
 
-        return interfaces
-
+        return addresses
 
 else:
 
-    def get_network_interfaces():
-        try:
-            return ifcfg.interfaces().values()
-        except:  # ifcfg throws an Exception if the ip/ifconfig commands are not found
-            return []
-
-
-def get_all_addresses():
-    return list(
-        set(iface["inet"] for iface in get_network_interfaces() if iface["inet"])
-    )
+    def get_all_addresses():
+        return list(set(addr.ip for iface in ifaddr.get_adapters() for addr in iface.ips if addr.is_IPv4))
```

Comment:
Do we imagine cases these days where there might only be IPv6? Guessing we're fine, but got me thinking.

Reply:
Newer upstream zeroconf has support for specifying whether you do IPv4, IPv6 or both. For simplicity I just used IPv4 here though.